core/vm/vmcontext/runner.go

```diff
@@ -0,0 +1,120 @@
+package vmcontext
+
+import (
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/core/state"
+	"github.com/celo-org/celo-blockchain/core/types"
+	"github.com/celo-org/celo-blockchain/core/vm"
+)
+
+// VMAddress is the address the VM uses to make internal calls to contracts
+var VMAddress = common.ZeroAddress
+
+// ExtendedChainContext extends ChainContext providing relevant methods
+// for EVMRunner creation
+type ExtendedChainContext interface {
+	vm.ChainContext
+
+	// GetVMConfig returns the node's vm configuration
+	GetVMConfig() *vm.Config
+
+	CurrentHeader() *types.Header
+
+	State() (*state.StateDB, error)
+}
+
+// SystemEVMRunnerBuilder returns a builder factory for a System's EVMRunner
```

Comment:
I don't like adding "builder" to the mix.  This is a function that returns a factory, so I would rename it to `GetSystemEVMRunnerFactory()`.  Then the name is more self-explanatory in the places it's used, e.g.

```
contract_comm.SetEVMRunnerFactory(vmcontext.GetSystemEVMRunnerFactory(blockchain))
```

Reply:
Done, marking as resolved.